src/path/json.rs

```diff
@@ -1,295 +1,300 @@
+//             if let Some(left_elems) = el.as_array() {
+//                 for l in left_elems.iter() {
+//                     for r in elems.iter() {
+//                         if l.eq(r) {
+//                             return true;
+//                         }
+//                     }
+//                 }
+//             } else {
+//                 for r in elems.iter() {
+//                     if el.eq(&r) {
+//                         return true;
+//                     }
+//                 }
+//             }
+//         }
+//     }
+
+//     false
+// }
+
+// /// ensure that the element on the left sides mathes the regex on the right side
+// pub fn regex<T>(left: Vec<&T>, right: Vec<&T>) -> bool {
+//     if left.is_empty() || right.is_empty() {
+//         return false;
+//     }
+
+//     match right.first() {
+//         Some(Value::String(str)) => {
+//             if let Ok(regex) = Regex::new(str) {
+//                 for el in left.iter() {
+//                     if let Some(v) = el.as_str() {
+//                         if regex.is_match(v) {
+//                             return true;
+//                         }
+//                     }
+//                 }
+//             }
+//             false
+//         }
+//         _ => false,
+//     }
+// }
+
+// /// ensure that the element on the left side belongs to the array on the right side.
+// pub fn inside<T>(left: Vec<&T>, right: Vec<&T>) -> bool {
+//     if left.is_empty() {
+//         return false;
+//     }
+
+//     match right.first() {
+//         Some(Value::Array(elems)) => {
+//             for el in left.iter() {
+//                 if elems.contains(el) {
+//                     return true;
+//                 }
+//             }
+//             false
+//         }
+//         Some(Value::Object(elems)) => {
+//             for el in left.iter() {
+//                 for r in elems.values() {
+//                     if el.eq(&r) {
+//                         return true;
+//                     }
+//                 }
+//             }
+//             false
+//         }
+//         _ => false,
+//     }
+// }
+
+// /// ensure the number on the left side is less the number on the right side
+// pub fn less<T>(left: Vec<&T>, right: Vec<&T>) -> bool {
+//     if left.len() == 1 && right.len() == 1 {
+//         match (left.first(), right.first()) {
+//             (Some(Value::Number(l)), Some(Value::Number(r))) => l
+//                 .as_f64()
+//                 .and_then(|v1| r.as_f64().map(|v2| v1 < v2))
+//                 .unwrap_or(false),
+//             _ => false,
+//         }
+//     } else {
+//         false
+//     }
+// }
+
+// /// compare elements
+// pub fn eq<T>(left: Vec<&T>, right: Vec<&T>) -> bool {
+//     if left.len() != right.len() {
+//         false
+//     } else {
+//         left.iter().zip(right).map(|(a, b)| a.eq(&b)).all(|a| a)
+//     }
+// }
+
+// #[cfg(test)]
+// mod tests {
```

Comment:
can you also move these tests to JsonLike simple for Value and use them there?
Maybe I overlooked it but I could not find where the tests were reimplemented